haystack_experimental/components/image_converters/image_utils.py

```diff
@@ -66,8 +70,11 @@ def _encode_image_to_base64(
     # NOTE: We prefer the format returned by PIL
     inferred_mime_type = image.get_format_mimetype() or bytestream.mime_type
 
-    # Downsize the image
-    downsized_image: "Image" = _resize_image_preserving_aspect_ratio(image, size)
+    # Downsize the image in place
+    if size is not None:
+        # Set reducing_gap=None to disable multi-step shrink; better quality.
+        # https://pillow.readthedocs.io/en/latest/reference/Image.html#PIL.Image.Image.thumbnail
+        image.thumbnail(size=size, reducing_gap=None)
```

Comment:
have to say good find again! Would not have guessed this resizing with aspect ratio preservation would be hidden behind a method called thumbnail :D